sphinx/builders/linkcheck.py

```diff
@@ -301,10 +316,15 @@ def write_doc(self, docname: str, doctree: Node) -> None:
         if self.broken:
             self.app.statuscode = 1
 
-    def write_entry(self, what: str, docname: str, line: int, uri: str) -> None:
-        with open(path.join(self.outdir, 'output.txt'), 'a', encoding='utf-8') as output:
-            output.write("%s:%s: [%s] %s\n" % (self.env.doc2path(docname, None),
-                                               line, what, uri))
+    def write_entry(self, what: str, docname: str, docpath: str, line: int,
+                    uri: str) -> None:
+        with open(path.join(self.outdir, 'output.txt'), 'a') as output:
+            output.write("%s:%s: [%s] %s\n" % (docpath, line, what, uri))
+
+    def write_jsonline(self, data: dict) -> None:
```

Comment:
I think jsonline is a name of data structure. We should name this like `write_linkstat()` to describe what this method doing. It also means we should give a good naming to this data.

Reply:
Changed to `write_linkstat()`